samples/client/petstore/python-experimental/petstore_api/model_utils.py

```diff
@@ -148,7 +148,7 @@ def __new__(cls, *args, **kwargs):
         oneof_anyof_classes = []
         oneof_anyof_classes.extend(cls._composed_schemas.get('oneOf', ()))
         oneof_anyof_classes.extend(cls._composed_schemas.get('anyOf', ()))
-        new_cls = cls.get_discriminator_class(kwargs)
+        new_cls = get_discriminator_class(cls, kwargs, set())
```

Comment:
We may be passing in _visited_composed_classes in kwargs here, so we need to include it in the inputs to get_discriminator_class.
Can you extract _visited_composed_classes from kwargs and pass it in to get_discriminator_class here?

Reply:
Never mind, I think that we are fine here

Reply:
I mean:
```
_visited_composed_classes = kwargs.get('_visited_composed_classes', ())
new_cls = get_discriminator_class(cls, kwargs, _visited_composed_classes)
```
You need this because if we travel through two discriminators. The `__new__` call to the class with the second discriminator will be passing in _visited_composed_classes here.

Reply:
hmm, ok, I am going through your code to try to understand what it does, because now there would be two parts of the code adding to the _visited_composed_classes tuple, with two separate tests to check if a class has already been visited or not.